test/core.jl

```diff
@@ -7213,6 +7213,20 @@ end
 @test_throws ArgumentError Array{Int, 2}(undef, -10, 0)
 @test_throws ArgumentError Array{Int, 2}(undef, -1, -1)
 
+# issue #54244
+# test that zero sized array doesn't throw even with large axes
+bignum = Int==Int64 ? 2^32 : 2^16
+Array{Int}(undef, 0, bignum, bignum)
+Array{Int}(undef, bignum, bignum, 0)
+Array{Int}(undef, bignum, bignum, 0, bignum, bignum)
+# but also test that it does throw if the axes multiply to a multiple of typemax(UInt)
+@test_throws ArgumentError Array{Int, 2}(undef, bignum, bignum)
+@test_throws ArgumentError Array{Int, 2}(undef, 1, bignum, bignum)
+# also test that we always throw erros for negative dims even if other dims are 0 or the product is positive
+@test_throws ArgumentError Array{Int, 2}(undef, 0, -4, -4)
+@test_throws ArgumentError Array{Int, 2}(undef, -4, 1, 0)
+@test_throws ArgumentError Array{Int, 2}(undef, -4, -4, 1)
```

Comment:
Matrices have 2 dimensions, not 3, maybe you wanted to use `Array{Int, 3}`?

Reply:
oops. That's why you don't blindly copy and paste.